File: backend/src/main/java/com/nigelle/ecommerceback/entities/Avis.java
package com.nigelle.ecommerceback.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Avis {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private int note;
    private String commentaire;

    @Temporal(TemporalType.TIMESTAMP)
    private Date dateAvis;
    @ManyToOne
    private Produit produit;
    @ManyToOne
    private AppUser user;
}

File: frontend/src/app/models/Avis.ts
import {Produit} from "./Produit";
import {AppUser} from "./AppUser";

export class Avis {

  id!: number;
  note!:number;
  commentaire!: string;
  dateAvis!:Date;
  produit!:Produit;
  user!:AppUser;
}
